vowpalwabbit/core/src/reductions/cb/details/large_action/one_pass_svd_impl.cc

```diff
@@ -71,8 +69,18 @@ void one_pass_svd_impl::generate_AOmega(const multi_ex& examples, const std::vec
 
       AOmega(row_index, col) = final_dot_prod * shrink_factors[row_index];
     }
+  };
+
+  uint64_t row_index = 0;
+  for (auto* ex : examples)
+  {
+    _futures.emplace_back(_thread_pool.submit(
+        calculate_aomega_row, row_index, p, _all, _seed, ex, std::ref(AOmega), std ::ref(shrink_factors)));
     row_index++;
   }
+
+  for (auto& ft : _futures) { ft.get(); }
```

Comment:
`calculate_aomega_row` could throw so we could wrap `ft.get()` in a try/catch and rethrow here (not sure how we would recover)